src/parser/Expression.ts

```diff
@@ -377,6 +377,21 @@ export class DottedGetExpression extends Expression {
             walk(this, 'obj', visitor, options);
         }
     }
+
+    getNameParts() {
```

Comment:
This function incorrectly assumes that all `DottedGetExpression` instances represent only dotted gets the whole way through. That's not how `DottedGetExpression` works. 

There's also already a util function that handles this logic: https://github.com/rokucommunity/brighterscript/blob/bc7367a570be3183db7c5889f526460e41de1a03/src/util.ts#L1377-L1381

The following expression contains `DottedGetExpression` instances mixed with `CallExpression`, `IndexedGetExpression`, and others. 
```
alpha.beta.charlie()["delta"]@.echo().foxtrot.golf
```

Here's a unit test proving this function explodes with the above expression.

```typescript
it('test getNameParts', () => {
    const file = program.setFile<BrsFile>('source/main.bs', `
        sub main()
            alpha = {}
            value = alpha.beta.charlie()["delta"]@.echo().foxtrot.golf
        end sub
    `);
    expectZeroDiagnostics(program);

    expect(
        file.ast.findChild<DottedGetExpression>(isDottedGetExpression).getNameParts()
    ).to.eql([]);
});
```

![image](https://user-images.githubusercontent.com/2544493/194309735-007fd4f5-bf16-40f7-9e35-93ab64c0ab9c.png)

Reply:
addressed. thanks